share/util/source_stat.py

```diff
@@ -0,0 +1,158 @@
+import logging
+
+import requests
+from furl import furl
+from lxml import etree
+import pendulum
+
+from share.models import SourceStat
+from share.models import SourceConfig
+
+logger = logging.getLogger(__name__)
+
+
+class SourceStatus():
+
+    def assert_no_exception(self, url, timeout=15.0):
+        try:
+            r = requests.get(url, timeout=timeout)
+        # except all exception and log
+        except Exception as e:
+            logger.warning('Exception received from source: %s', e)
+            return (None, e)
+        return (r, None)
+
+    def get_source_stats(self, config_id):
+        sourceconfig = SourceConfig.objects.get(pk=config_id)
```

Comment:
As the entire class is built around the source config, it might be a bit nicer to just pass it into `__init__` and then just reference it as `self.source_config` or whatever name you prefer.